packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -1,5 +1,5 @@
 <script lang="ts" setup>
-import { computed } from "vue";
+import { computed, ref } from "vue";
 
 const props = withDefaults(
```

Comment:
To be discussed:
1. we planned to export the prop types of all components within the component (if I remember correctly)
2. it would be better to somehow refer to the native html component's props when typing the props of this component.

So I tried the following:
```vue
type NativeInputProps = Pick<
  HTMLInputElement,
  "required" | "minLength" | "maxLength" | "min" | "max" | "type" | "pattern" | "title"
>;
export type TestInputProps = {
  /** The current input value */
  modelValue?: string;
  /** Label to show next to the input */
  label?: string;
} & Partial<NativeInputProps>;

const props = withDefaults(defineProps<TestInputProps>(), {
  modelValue: "",
  label: "",
  type: "text",
});
```

But the vue compiler really does not like it...

![image](https://github.com/SchwarzIT/onyx/assets/151019977/b07b1058-53ef-4ac0-84fa-90581ea123cb)


not sure if it's because of the union type or because it gets overwhelmed by the >350 properties that the HTMLInputElement has... 🤔

Reply:
seems to be the latter, just using the union type seems to work properly.

Reply:
doing it properly wihtout the pick gets a bit long ... did we want to move the exported types outside of the component files?

```ts
type HTMLInputTypeAttribute =
  | "button"
  | "checkbox"
  | "color"
  | "date"
  | "datetime-local"
  | "email"
  | "file"
  | "hidden"
  | "image"
  | "month"
  | "number"
  | "password"
  | "radio"
  | "range"
  | "reset"
  | "search"
  | "submit"
  | "tel"
  | "text"
  | "time"
  | "url"
  | "week";
  /** equivalent to Pick<HTMLInputElement,
 * "required" | "minLength" | "maxLength" | "min" | "max" | "type" | "pattern" | "title"
 * >;
 */
type NativeInputProps = {
  required?: boolean | undefined;
  pattern?: string | undefined;
  type?: HTMLInputTypeAttribute | undefined;
  max?: number | string | undefined;
  maxLength?: number | undefined;
  min?: number | string | undefined;
  minLength?: number | undefined;
};
export type TestInputProps = {
  /** The current input value */
  modelValue?: string;
  /** Label to show next to the input */
  label?: string;
} & Partial<NativeInputProps>;

const props = withDefaults(defineProps<TestInputProps>(), {
  modelValue: "",
  label: "",
  type: "text",
});
```

Reply:
tried to condence this now. but in the end, we should definitely declare which "type" values are allowed, but will probably have separate components for some of those types anyways :)

Reply:
Yes I don't think it makes sense to allow types such as date, file, time etc. when we have dedicated datepickers, filepickers etc.

Reply:
done. let's still discuss tomorrow if we can make it work somehow to connect to the HTMLInputTypeAttribute interface without making vue break (couldn't find it out so far)
and where we want to keep the prop type definitions -> in each .vue file or in (a) separate file(s).

Reply:
1. for now, we keep the types it in the .vue file.
2. created a follow up task for investigating the compiler crash. https://github.com/orgs/SchwarzIT/projects/5/views/7?filterQuery=task-priority%3Adev&pane=issue&itemId=48984776

Reply:
This issue is not only a compiler crash. I noticed that Storybook currently does not support generating docs for props when you use a union type such as `export type TestInputProps = { //... } & Partial<NativeInputProps>`

So they will not be documented which is a problem